test/downstream/integrator_indexing.jl

```diff
@@ -367,3 +367,19 @@ setp(sys, p)(integrator, [4, 5, 6])
 @test getp(sys, p)(integrator) == integrator.ps[p] == [4, 5, 6]
 integrator.ps[p] = [7, 8, 9]
 @test getp(sys, p)(integrator) == integrator.ps[p] == [7, 8, 9]
+
+# Issue#653
+@parameters p
+@variables X(t)
+
+eq = D(X) ~ p - X
+@mtkbuild osys = ODESystem([eq], t)
+oprob = ODEProblem(osys, [X => 0.1], (0.0, 1.0), [p => 1.0])
+
+oint = init(oprob, Tsit5())
+@test_throws ErrorException oint[X]=1.0
+@test_throws ErrorException setu(oint, X)(oint, 10.0)
```

Comment:
Why would this error?

Reply:
https://github.com/SciML/OrdinaryDiffEq.jl/blob/258bbbbc7458b66c14fe4a03054252d176c9dea8/src/integrators/integrator_interface.jl#L480-L481

Reply:
Changing out the `u` vector is different from mutating it. These functions should just mutate `u`.

Reply:
Fixed now